pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -116,20 +118,52 @@ func (moa *MultiOrgAlertmanager) LoadAndSyncAlertmanagersForOrgs(ctx context.Con
 
 	// Then, sync them by creating or deleting Alertmanagers as necessary.
 	moa.metrics.DiscoveredConfigurations.Set(float64(len(orgIDs)))
-	moa.SyncAlertmanagersForOrgs(orgIDs)
+	moa.SyncAlertmanagersForOrgs(ctx, orgIDs)
 
 	moa.logger.Debug("done synchronizing Alertmanagers for orgs")
 
 	return nil
 }
 
-func (moa *MultiOrgAlertmanager) SyncAlertmanagersForOrgs(orgIDs []int64) {
+//getLatestConfigs retrieves the latest Alertmanager configuration for every organization and returns them as a map where the key is ID of an organization
+func (moa *MultiOrgAlertmanager) getLatestConfigs(ctx context.Context) (map[int64]*models.AlertConfiguration, error) {
+	configs, err := moa.configStore.GetAllLatestAlertmanagerConfiguration(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[int64]*models.AlertConfiguration, len(configs))
+	for _, config := range configs {
+		result[config.OrgID] = config
+	}
+
+	return result, nil
+}
+
+func (moa *MultiOrgAlertmanager) SyncAlertmanagersForOrgs(ctx context.Context, orgIDs []int64) {
 	orgsFound := make(map[int64]struct{}, len(orgIDs))
+	dbConfigs, err := moa.getLatestConfigs(ctx)
+	if err != nil {
+		moa.logger.Error("failed to load Alertmanager configurations", "err", err)
+		return
+	}
 	moa.alertmanagersMtx.Lock()
 	for _, orgID := range orgIDs {
 		orgsFound[orgID] = struct{}{}
 
-		existing, found := moa.alertmanagers[orgID]
+		dbConfig, cfgFound := dbConfigs[orgID]
+
+		var cfg *api.PostableUserConfig
+		if cfgFound {
```

Comment:
It did not feel right to me that DTO leaks into Alertmanager's domain. There are methods of Alertmanager that implement access to database, which in my opinion should not belong there but be a separate access layer (perhaps a repository). I am ok with your suggestion.

Reply:
I was about to implement the suggestion and then noticed that none of the Alertmanager's methods accept AlertmanagerConfiguration but many accept api.PostableUserConfig. So I still think that the parsing should be done in upstream.

Reply:
The ones that accept `api.PostableUserConfig` are the methods that get called directly from the API - because there's nothing we can parse from/to, after all, it is the direct user input. 

There are three uses of `Load` in this codebase, all three were directly in `notifier/alertmanager.go` by moving this one here (whether is correct or not is a separate conversation) we've just made inconsistent with the rest and thus harder to change. I'd strongly prefer if move it back, then if you have a strong preference for it you can create a separate PR where you can find a place for all 3 😄.